### ISSUE TITLE
object_store_logger._validate_credentials needs retries

### ISSUE THREAD

**Issue by moinnadeem** [mosaicml/composer#1105]
When a user launches a large sweep (10 jobs), and all jobs start immediately, then `_validate_credentials` is called on all of the nodes at the same time. Occasionally, this results in a `429: Too Many Requests` from the object store provider, causing all of the jobs to fail.

Similarly to how we handle transient errors with exponential backoffs for uploading artifacts, we should handle transient errors with exponential backoffs for [validating credentials](https://github.com/mosaicml/composer/blob/356ac4371e7077738088decc9d9e6db77bb7886e/composer/loggers/object_store_logger.py#L373) too.

**Comment by ravi-mosaicml** [comment]
Good point; in addition to retrying on transient errors, thoughts about only validating credentials on global rank 0 (or local rank 0)?

**Comment by moinnadeem** [comment]
> thoughts about only validating credentials on global rank 0 (or local rank 0)?

Oh, no wonder I was seeing 429's, we had roughly 96 requests sent within seconds of each other, causing a cascading set of jobs to fail.

Agreed, the PR should make two changes:
1. Retrying on transient errors
2. Validate credentials only on rank 0
